storage/factory/pruningStorerFactory.go

```diff
@@ -329,13 +348,19 @@ func (psf *StorageServiceFactory) CreateForMeta() (dataRetriever.StorageService,
 	}
 
 	userAccountsUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.AccountsTrieStorage, customDatabaseRemover)
-	userAccountsUnit, err = psf.createTriePruningPersister(userAccountsUnitArgs)
+	if psf.storageType == ProcessStorageService {
```

Comment:
a lot of duplicated code. Let's refactor. Also, this if is pretty ugly

In case we do not have psf.storageType == ProcessStorageService do we let the userAccountsUnitArgs.PersistersTracker nil?
Why didn't we have created a method factory?

Reply:
UserAccountsUnitArgs.PersistersTracker is initialized in createPruningStorerArgs, and is just replaced here, so it is never nil. Refactored to remove duplicated code.